test/hotspot/jtreg/runtime/cds/appcds/SharedArchiveConsistency.java

```diff
@@ -276,23 +274,38 @@ public static void modifyHeaderIntField(long offset, int value) throws Exception
         }
     }
 
-    public static void copyFile(File from, File to) throws Exception {
-        if (to.exists()) {
-            if(!to.delete()) {
-                throw new IOException("Could not delete file " + to);
+    static int testCount = 0;
+    public static String startNewTestArchive(String testName) {
+        ++ testCount;
+        String newArchiveName = TestCommon.getNewArchiveName(String.format("%02d", testCount) + "-" + testName);
+        TestCommon.setCurrentArchiveName(newArchiveName);
+        return newArchiveName;
+    }
+
+    public static File copyFile(File orgJsaFile, String testName) throws Exception {
+        File newJsaFile = new File(startNewTestArchive(testName));
+        if (newJsaFile.exists()) {
+            if (!newJsaFile.delete()) {
+                throw new IOException("Could not delete file " + newJsaFile);
             }
         }
-        to.createNewFile();
-        setReadWritePermission(to);
-        Files.copy(from.toPath(), to.toPath(), REPLACE_EXISTING);
+        newJsaFile.createNewFile();
+        Files.copy(orgJsaFile.toPath(), newJsaFile.toPath(), REPLACE_EXISTING);
```

Comment:
Not clear why you need to create the file if you are going to immediately replace it.

Reply:
I removed the call to createNewFile and it seems to work just fine. I don't remember why the call was there before. I'll do more testing before integrating.